pkg/mutation/schema/schema.go

```diff
@@ -89,7 +89,12 @@ func (db *DB) upsert(mutator MutatorWithSchema) error {
 
 	path := mutator.Path()
 	bindings := mutator.SchemaBindings()
-	db.cachedMutators[id] = mutator.DeepCopy().(MutatorWithSchema)
+	var ok bool
+	mutatorCopy := mutator.DeepCopy()
+	db.cachedMutators[id], ok = mutatorCopy.(MutatorWithSchema)
+	if !ok {
+		panic(fmt.Sprintf("got mutator.DeepCopy() type %T, want %T", mutatorCopy, MutatorWithSchema(nil)))
```

Comment:
nit: the linting change here seems superfluous, as we are still panicking. One day Golang generics will save us.

Reply:
Yeah, all this location does for now is present a more human-readable error. Agreed - generics will make this sort of problem go away.